ARCH.md

```diff
@@ -0,0 +1,55 @@
+Rust-Lightning is broken into a number of high-level structures with APIs to hook them
+together, as well as APIs for you, the user, to provide external data.
+
+The two most important structures which nearly every application of Rust-Lightning will
+need to use are `ChannelManager` and `ChannelMonitor`. `ChannelManager` holds multiple
+channels, routes payments between them and exposes a simple API to make and receive
+payments. Individual `ChannelMonitor`s monitor the on-chain state of a channel, punish
+counterparties if they misbehave, and force-close channels if they contain unresolved
+HTLCs which are near expiration. The `ManyChannelMonitor` API provides a way for you to
+receive `ChannelMonitorUpdate`s from `ChannelManager` and persist them to disk before the
+channel steps forward.
+
+There are several additional high-level structures that you may use either on the same
+device as the `ChannelManager` or on a separate one. They include `Router`, which handles
+receiving channel and node announcements, and calculates routes for sending payments.
+`PeerManager` handles the authenticated and encrypted communication protocol, monitoring
+for liveness of peers via `timer_tick_occured()` and routes messages to/from a
+`ChannelManager` and `Router` instance as appropriate.
```

Comment:
Hmm, yea, not sure how to clarify that, cause its even more subtle than that, too - You're somewhat expected to poll [PeerManager::process_events](https://docs.rs/lightning/0.0.10/lightning/ln/peer_handler/struct.PeerManager.html#method.process_events), which *will* fetch messages and send them outbound. Let me know how you think that could be clarified.

Reply:
Let's leave out the detail on polling as it gets a bit too much into how `PeerManager` should be used rather than sticking with its function.
For the subtlety that I raised, perhaps adding something like the following sentence would suffice:
> Outgoing messages produced by `ChannelManager` are communicated to `PeerManager` using an `EventsProvider`.
Or something along those lines that ties it back to the diagram. Feel free to scrap if it sounds a bit too detailed. I'd be fine without it.